src/client/renderer/chunk_renderer.cpp

```diff
@@ -120,18 +137,36 @@ void ChunkRenderer::updateMesh(const ChunkPosition& position,
 ChunkRenderResult ChunkRenderer::renderChunks(const glm::vec3& cameraPosition,
                                               const ViewFrustum& frustum,
                                               const glm::mat4& projectionViewMatrix,
-                                              bool cameraInWater)
+                                              bool cameraInWater, int renderDistance)
 {
     auto& solidDrawables = m_chunkRenderables[static_cast<int>(ChunkRenderType::Solid)];
     auto& fluidDrawables = m_chunkRenderables[static_cast<int>(ChunkRenderType::Fluid)];
     auto& floraDrawables = m_chunkRenderables[static_cast<int>(ChunkRenderType::Flora)];
 
-    for (auto& meshes : m_chunkMeshes) {
-        bufferChunks(meshes.voxelMesh, solidDrawables);
-        bufferChunks(meshes.fluidMesh, fluidDrawables);
-        bufferChunks(meshes.floraMesh, floraDrawables);
+    // Goes through unrendered chunks and render them if in range
+    for (unsigned int i = 0; i < m_chunkMeshes.size(); i++) {
+        if (::isChunkInRenderDistance(cameraPosition, m_chunkMeshes[i].voxelMesh.position,
+                                      renderDistance)) {
+            bufferChunks(m_chunkMeshes[i].voxelMesh, solidDrawables);
+            bufferChunks(m_chunkMeshes[i].fluidMesh, fluidDrawables);
+            bufferChunks(m_chunkMeshes[i].floraMesh, floraDrawables);
+            m_chunkMeshesRendered.push_back(m_chunkMeshes[i]);
+            m_chunkMeshes.erase(m_chunkMeshes.begin() + i);
+            i--;
+        }
+    }
+
+    // Goes through rendered chunks and unrender them if not in range
+    for (unsigned int i = 0; i < m_chunkMeshesRendered.size(); i++) {
+        if (!::isChunkInRenderDistance(cameraPosition,
+                                       m_chunkMeshesRendered[i].voxelMesh.position,
+                                       renderDistance)) {
+            deleteChunkRenderables(m_chunkMeshesRendered[i].voxelMesh.position);
+            m_chunkMeshes.push_back(m_chunkMeshesRendered[i]);
+            m_chunkMeshesRendered.erase(m_chunkMeshesRendered.begin() + i);
+            i--;
+        }
```

Comment:
Not too sure if this is needed for now.

I think having the `isChunkInRenderDistance` function called from the other [renderChunks](https://github.com/Hopson97/open-builder/blob/dfe54bb5e5581bfb9456c5072fb4b013eeeba92f/src/client/renderer/chunk_renderer.cpp#L50) function is sufficient for now.